gitmanager.py

```diff
@@ -306,20 +306,56 @@ def remove_from_blacklist(cls, item, username, blacklist_type="", code_privilege
     @classmethod
     def merge_pull_request(cls, pr_id, comment=""):
         response = requests.get("https://api.github.com/repos/{}/pulls/{}".format(GlobalVars.bot_repo_slug, pr_id))
-        if not response:
+        files_list = requests.get("https://api.github.com/repos/{}/pulls/{}/files"
+                                  .format(GlobalVars.bot_repo_slug, pr_id))
+        if not response or not files_list:
             raise ConnectionError("Cannot connect to GitHub API")
         pr_info = response.json()
+        files_info = files_list.json()
         if pr_info["user"]["login"] != "SmokeDetector":
             raise ValueError("PR #{} is not created by me, so I can't approve it.".format(pr_id))
         if "<!-- METASMOKE-BLACKLIST" not in pr_info["body"]:
             raise ValueError("PR description is malformed. Blame a developer.")
         if pr_info["state"] != "open":
             raise ValueError("PR #{} is not currently open, so I won't merge it.".format(pr_id))
+        pr_string = pr_info["title"]
+        file_modified = files_info[0]["filename"]
         ref = pr_info['head']['ref']
+        requested_pr_string = regex.match(r".*?: \S+ (.*?)(?:\Z|\s*\(\?#)", pr_string).group(1)
+        username_of_pr_requester = regex.match(r".*(?=:)", pr_string).group(0)
+        if file_modified == "blacklisted_websites.txt":
+            blacklist_type = Blacklist.WEBSITES
+        elif file_modified == "bad_keywords.txt":
+            blacklist_type = Blacklist.WEBSITES
+        elif file_modified == "blacklisted_usernames.txt":
+            blacklist_type = Blacklist.USERNAMES
+        elif file_modified == "blacklisted_numbers.txt":
+            blacklist_type = Blacklist.NUMBERS
+        elif file_modified == "watched_keywords.txt":
+            blacklist_type = Blacklist.WATCHED_KEYWORDS
+        elif file_modified == "watched_numbers.txt":
+            blacklist_type = Blacklist.WATCHED_NUMBERS
+        else:
+            raise CmdException('GitManager: blacklist is not recognized. Blame a developer.')
+        try:
+            blacklister = Blacklist(blacklist_type)
+            blacklist_file_name = blacklist_type[0]
+            item_to_blacklist = requested_pr_string
+            if blacklist_type in {Blacklist.WATCHED_KEYWORDS, Blacklist.WATCHED_NUMBERS}:
+                op = 'watch'
+                item = item_to_blacklist
+                item_to_blacklist = "\t".join([now, username_of_pr_requester, item])
```

Comment:
What's `now`? Presumably it's the current time, but I don't see it declared anywhere?

Reply:
This cannot be tested due to my test instance not being able to get the latest blacklist information, therefore testing the watched string will always result in it being not caught therefore the testing will be inaccurate. **This will not work.** @makyen  Identified some issues with the RegEx a couple of months ago (see previous comment) but has not gotten to working on it yet/forgot about it. It does seem like the time is missing, when Makyen fixes the RegEx I'll look into it.

Reply:
`now` needs to be defined per-method, this is defined in other methods for blacklisting, etc. but not here.  Fixing the non-declaration of `now` is trivial, ultimately.  See line 113 in this file (another method's declaration) for how it's done.

Reply:
@teward I already responded to this, see above.

Reply:
> @teward I already responded to this, see above.

I saw your response.  I'm still making my point known.

Reply:
It's unclear to me how anything I've said I'll work on will affect this specific concern with this code. Please explain. As you've said, I must have forgotten about something. Please help me by reminding me.

Reply:
@makyen You said that you needed to change the RegEx, I can't remember what exactly as the conversation is lost.